pontoon/base/models.py

```diff
@@ -3427,6 +3456,25 @@ def calculate_stats(self, save=True):
         resource = self.resource
         locale = self.locale
 
+        if self.resource.project.visibility == "private":
+            if save:
+                self.adjust_all_stats(
+                    -self.total_strings,
+                    -self.approved_strings,
+                    -self.fuzzy_strings,
+                    -self.strings_with_errors,
+                    -self.strings_with_warnings,
+                    -self.unreviewed_strings,
+                )
+            else:
+                self.total_strings = 0
+                self.approved_strings = 0
+                self.fuzzy_strings = 0
+                self.strings_with_errors = 0
+                self.strings_with_warnings = 0
+                self.unreviewed_strings = 0
+            return
```

Comment:
I think this is an interesting approach, but it has a few drawbacks:
1. If the changes are made to translations of a private project (e.g. by an admin), stats will be stored again (and not stay at 0).
1. Users who see private projects should see the actual stats, not 0.
1. If you flip the visibility switch for an existing (partially translated) project (which is the use case we'll need as early as next week if we ship the feature in time), stats don't update correctly right now.

Instead I propose we revert the changes to the stats processing you have made in the last two commits and treat private projects similarly as `disabled` and `system_projects`:
1. When the [visibility status changes](https://github.com/mozilla/pontoon/pull/1619/files?file-filters%5B%5D=.css&file-filters%5B%5D=.py#diff-51dc0e413b1d5b6b4a182d2e14ab4909R62), run `locale.aggregate_stats()` for each project-locale.
1. In [`Locale.aggregate_stats()`](https://github.com/mozilla/pontoon/blob/master/pontoon/base/models.py#L1617), only include `TranslatedResources` with `resource__project__visibility="public"`.
1. In [`TranslatedResourceQuerySet.stats()`](https://github.com/mozilla/pontoon/blob/master/pontoon/base/models.py#L3323): if `project.slug == "all-projects"`, only include `TranslatedResources` with `resource__project__visibility="public"`.